src/formatter/index.ts

```diff
@@ -176,35 +176,25 @@ export class QueryFormatter {
     // YYYY-MM-DD HH:mm:ss.mmm
     const yearMonthDay =
       zeroPad(year, 4) + "-" + zeroPad(month, 2) + "-" + zeroPad(day, 2);
-    const hourMinuteSecond =
-      zeroPad(hour, 2) + ":" + zeroPad(minute, 2) + ":" + zeroPad(second, 2);
+    const time =
+      zeroPad(hour, 2) +
+      ":" +
+      zeroPad(minute, 2) +
+      ":" +
+      zeroPad(second, 2) +
+      "." +
+      zeroPad(millisecond, 3);
 
     if (param instanceof PGDate) {
       return this.escapeString(yearMonthDay);
     }
 
     if (param instanceof TimestampTZ) {
-      const str =
-        yearMonthDay +
-        " " +
-        hourMinuteSecond +
-        "." +
-        zeroPad(millisecond, 6, "right") +
-        " " +
-        param.timeZone;
+      const str = yearMonthDay + " " + time + " " + param.timeZone;
       return this.escapeString(str);
     }
 
-    if (param instanceof TimestampNTZ) {
```

Comment:
I would like to not serialize miliseconds for date type 
queries like this: `select '2022-05-04 17:37:19.10'::date`
will not work as date type do not support fraction of second

Reply:
@Jeseeq I'd prefer to be able to have all of the available time information in the JS `Date` object to be serialized if possible. Would you accept serialization of milliseconds if the ms part is not zero? That could help a bit with the `::DATE` casting issue.

Otherwise another way to work around the DATE casing would be to do `::TIMESTAMP::DATE`.

Reply:
Agree 
Then we can use something like:
```
const time =
      zeroPad(hour, 2) +
      ":" +
      zeroPad(minute, 2) +
      ":" +
      zeroPad(second, 2) +
      millisecond !== 0 ? ("."  + zeroPad(millisecond, 3)) : "";
```

Reply:
Great! I'll update this PR.